assets/dev/js/editor/components/template-library/manager.js

```diff
@@ -297,15 +297,28 @@ TemplateLibraryManager = function() {
 	};
 
 	this.loadTemplates = function( onUpdate ) {
-		self.requestLibraryData( {
-			onBeforeUpdate: self.layout.showLoadingView.bind( self.layout ),
-			onUpdate: function() {
-				self.layout.hideLoadingView();
+		self.layout.showLoadingView();
 
-				if ( onUpdate ) {
-					onUpdate();
-				}
-			},
+		const query = { source: this.getFilter( 'source' ) },
+			options = {};
+
+		// TODO: Remove - it when all the data commands is ready, manage the cache!.
+		if ( 'local' === query.source ) {
+			options.refresh = true;
+		}
+
+		$e.data.get( 'library/templates', query, options ).then( ( result ) => {
```

Comment:
Maybe should return the `promise` and remove the `onUpdate` method. but it should be checked

Reply:
Will be out of scope.